src/Arcus.Observability.Telemetry.AspNetCore/Arcus.Observability.Telemetry.AspNetCore.csproj

```diff
@@ -0,0 +1,38 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>netstandard2.0;netcoreapp3.1</TargetFrameworks>
+    <PackageId>Arcus.Observability.Telemetry.AspNetCore</PackageId>
+    <Authors>Arcus</Authors>
+    <Company>Arcus</Company>
+    <Description>Provides capability to improve ASP.NET Core telemetry with Serilog in applications</Description>
+    <Copyright>Copyright (c) Arcus</Copyright>
+    <PackageLicenseUrl>https://github.com/arcus-azure/arcus.observability/blob/master/LICENSE</PackageLicenseUrl>
+    <PackageProjectUrl>https://github.com/arcus-azure/arcus.observability</PackageProjectUrl>
+    <PackageIconUrl>https://raw.githubusercontent.com/arcus-azure/arcus/master/media/arcus.png</PackageIconUrl>
+    <RepositoryUrl>https://github.com/arcus-azure/arcus.observability</RepositoryUrl>
+    <RepositoryType>Git</RepositoryType>
+    <PackageTags>Azure;Observability;Telemetry;Serilog;ASP.NET;Core</PackageTags>
+    <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
+    <GenerateDocumentationFile>true</GenerateDocumentationFile>
+    <AssemblyName>Arcus.Observability.Telemetry.AspNetCore</AssemblyName>
+    <RootNamespace>Arcus.Observability.Telemetry.AspNetCore</RootNamespace>
+  </PropertyGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp3.1'">
+    <FrameworkReference Include="Microsoft.AspNetCore.App" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetFramework)' != 'netcoreapp3.1'">
```

Comment:
Wouldn't it be better to explicitly check on netstandard2 here ?
Because, what if we're using it in .NET 5 ?  Don't know what will happen then ?  I assume that .NET 5 also depends on Microsoft.AspnetCore.App, but I haven't played around with it.

Reply:
>  I assume that .NET 5 also depends on Microsoft.AspnetCore.App,
I think so too: https://github.com/dotnet/core/blob/master/release-notes/5.0/5.0.1/5.0.1.md

OK, so we should indeed check for `netstandard` so we can later move to .NET 5.

Reply:
I also updated the correlation project bc it had the same conditional check there.

Reply:
Sorry for me to be a bit picky on this, but I'm not convinced that the changes that have been done now, are good.  

What frameworks are we supporting in Arcus ? 
I'd say that if we support .NET 5.0, .NET Core 3.x and .NET standard, that we should maybe explicitly test on those.

For instance :
```csharp
<ItemGroup Condition = "'$(TargetFramework)' = 'net5.0' or '$(TargetFramework)' = 'netcoreapp3.1'">
```
and 
```csharp
<ItemGroup Condition="'$(TargetFramework)' = 'netstandard2.0'>
```

These are assumptions based on [this](https://docs.microsoft.com/en-us/dotnet/standard/frameworks#supported-target-frameworks) article.

(Don't know if this works, 'cause I haven't tested it or played with it).

Reply:
Related:  should we explicitly support .NET 5 ?  @tomkerkhove  ?

Reply:
I was going to try to use the preview build on myget in a .NET 5 app to see what it gives, but it looks like there's no preview for observability available on myget ?

Reply:
> Related: should we explicitly support .NET 5 ? @tomkerkhove ?

Nothing planned yet but feel free to open an issue since we should start supporting it



> I was going to try to use the preview build on myget in a .NET 5 app to see what it gives, but it looks like there's no preview for observability available on myget ?

You should see it though, see https://dev.azure.com/codit/Arcus/_build/results?buildId=50261&view=logs&jobId=02805f64-2152-5b7a-07d4-706789a8a527&j=02805f64-2152-5b7a-07d4-706789a8a527&t=79e516e3-a0d7-5428-bbee-4af2fc23326b

Reply:
Ah, I had to check the 'preview' checkbox.

But, it's not clear to me which package I need to use now if I want to make use of this library when I have no dependency on AspNetCore.
These are the packages that are available:

![image](https://user-images.githubusercontent.com/3605786/103277463-54821f00-49c9-11eb-833f-977704dc9c67.png)

I see that there's a Arcus.Observability.Telemetry.AspNetCore package, which I don't want to use.  This is the one that needs to be used in Api projects or other projects that depend on aspnet core.
The package Arcus.Observability.Telemetry.Serilog package seems to be an old one, so it isn't that one either I guess.

Is it ....Sinks.ApplicationInsights ?

Next to that: are the Telemetry.IoT and Telemetry.Sql packages enrichers ? If so, why aren't they called `Telemetry.Serilog.Enrichers.IoT` and `Telemetry.Serilog.Enrichers.Sql` ?  
Why do some packages have Serilog in their name, and others don't ?  It looks like all packages depend on Serilog ?

Reply:
> But, it's not clear to me which package I need to use now if I want to make use of this library when I have no dependency on AspNetCore.

This PR fix is about removing the ASP.NET Core dependency from our general Core package. So that's the `Arcus.Observability.Telemetry.Core` package.

> I see that there's a Arcus.Observability.Telemetry.AspNetCore package, which I don't want to use. This is the one that needs to be used in Api projects or other projects that depend on aspnet core.

Yes, this package is about tracking dependencies directly from `HttpRequest`/`HttpResponse` types without extracting the valuable information yourself.

> The package Arcus.Observability.Telemetry.Serilog package seems to be an old one, so it isn't that one either I guess.

Yes that's an old one. This is now split into multiple package, each with a different purpose (enrichers, sinks...)

 > Is it ....Sinks.ApplicationInsights ?

This package is about the Application Insights implementation of a Serilog sink. Not directly related to telemetry tracking itself.

> Next to that: are the Telemetry.IoT and Telemetry.Sql packages enrichers ? 

No, these packages are about tracking dependencies with specific IoT, Sql, AspNetCore... types. The reason they are in separate packages is because they require additional NuGet packages and we don't want them in our general telemetry core library.

>  If so, why aren't they called Telemetry.Serilog.Enrichers.IoT and Telemetry.Serilog.Enrichers.Sql ?

As answered above, they're not enricher packages and are not related to anything about Serilog.

> Why do some packages have Serilog in their name, and others don't ?

The packages with 'Serilog' in their name is in indication that these libraries are extensions on Serilog (enrichers, sinks, ...). The packages without the 'Serilog' in their name is about telemetry tracking but has in fact nothing to do with Serilog directly. 

>  It looks like all packages depend on Serilog ?

No, that's no true. The packages without 'Serilog' in their name is about general telemetry tracking. It can be used outside an Serilog environment.

Reply:
> Sorry for me to be a bit picky on this, but I'm not convinced that the changes that have been done now, are good.
> 
> What frameworks are we supporting in Arcus ?
> I'd say that if we support .NET 5.0, .NET Core 3.x and .NET standard, that we should maybe explicitly test on those.
> 
> For instance :
> 
> ```cs
> <ItemGroup Condition = "'$(TargetFramework)' = 'net5.0' or '$(TargetFramework)' = 'netcoreapp3.1'">
> ```
> 
> and
> 
> ```cs
> <ItemGroup Condition="'$(TargetFramework)' = 'netstandard2.0'>
> ```
> 
> These are assumptions based on [this](https://docs.microsoft.com/en-us/dotnet/standard/frameworks#supported-target-frameworks) article.
> 
> (Don't know if this works, 'cause I haven't tested it or played with it).

@tomkerkhove , I think it's better to take this up in another issue (maybe even on the general Arcus repo) so that we can discus that outside this issue. Since this PR is about removing the ASP.NET Core dependency and has as of now nothing to do (yet) with an upcoming .NET version.

Reply:
I agree to have another look at it via another issue / PR.   Current implementation might be sufficient for now.

Reply:
Created: https://github.com/arcus-azure/arcus/issues/106